### ISSUE TITLE
/etc/hosts not being rewritten when using Weave ECS AMI

### ISSUE THREAD

**Issue by bboreham** [weaveworks/weave#3322]
Observed by a user: 

(inside a container)
```
root@redis:/data# cat /etc/hosts
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
172.17.0.2      redis.weave.local redis
```

The issues are it doesn't say `# created by Weave` and the address at the end is the Docker address.

No error message observed; it works when I try it on my laptop.

Weave logs:
```
INFO: 2018/06/12 17:12:02.488616 Command line options: map[ipalloc-range:10.32.0.0/12 docker-bridge:docker0 host-root:/host http-addr:127.0.0.1:6784 nickname:ip-172-31-0-170 status-addr:127.0.0.1:6782 datapath:datapath H:[unix:///var/run/weave/weave.sock] hostname-from-label:com.amazonaws.ecs.container-name proxy:true resolv-conf:/var/run/weave/etc/resolv.conf dns-effective-listen-address:172.17.0.1 plugin:false port:6783 weave-bridge:weave dns-listen-address:172.17.0.1:53]
INFO: 2018/06/12 17:12:02.488960 weave  2.0.4
INFO: 2018/06/12 17:12:02.492466 Docker API on unix:///var/run/docker.sock: &[Arch=amd64 KernelVersion=4.9.43-17.38.amzn1.x86_64 GoVersion=go1.7.5 Os=linux MinAPIVersion=1.12 GitCommit=7392c3b/17.03.2-ce BuildTime=2017-08-09T22:45:09.101301574+00:00 Version=17.03.2-ce ApiVersion=1.27]
INFO: 2018/06/12 17:12:02.493302 Using docker bridge IP for DNS: 172.17.0.1
INFO: 2018/06/12 17:12:02.498220 proxy listening on unix:///var/run/weave/weave.sock
INFO: 2018/06/12 17:12:02.728856 Bridge type is bridged_fastdp
INFO: 2018/06/12 17:12:02.728947 Communication between peers is unencrypted.
INFO: 2018/06/12 17:12:02.743621 Our name is 66:fb:9a:bf:54:f0(ip-172-31-0-170)
INFO: 2018/06/12 17:12:02.743707 Launch detected - using supplied peer list: [172.31.0.61 172.31.0.72]
INFO: 2018/06/12 17:12:02.744751 Docker API on unix:///var/run/docker.sock: &[KernelVersion=4.9.43-17.38.amzn1.x86_64 BuildTime=2017-08-09T22:45:09.101301574+00:00 Version=17.03.2-ceApiVersion=1.27 MinAPIVersion=1.12 GoVersion=go1.7.5 Arch=amd64 GitCommit=7392c3b/17.03.2-ce Os=linux]
INFO: 2018/06/12 17:12:02.745205 Checking for pre-existing addresses on weave bridge
INFO: 2018/06/12 17:12:02.789267 [allocator 66:fb:9a:bf:54:f0] No valid persisted data
INFO: 2018/06/12 17:12:02.797754 [allocator 66:fb:9a:bf:54:f0] Initialising via deferred consensus
INFO: 2018/06/12 17:12:02.803146 Listening for DNS queries on 172.17.0.1
INFO: 2018/06/12 17:12:02.815165 Sniffing traffic on datapath (via ODP)
INFO: 2018/06/12 17:12:02.819538 Listening for HTTP control messages on 127.0.0.1:6784
INFO: 2018/06/12 17:12:02.820398 Listening for metrics requests on 127.0.0.1:6782
INFO: 2018/06/12 17:12:02.842989 ->[172.31.0.72:6783] attempting connection
INFO: 2018/06/12 17:12:02.843229 ->[172.31.0.61:6783] attempting connection
INFO: 2018/06/12 17:12:02.844972 ->[172.31.0.72:6783|2e:d4:1b:0c:6d:d5(ip-172-31-0-72)]: connection ready; using protocol version 2
INFO: 2018/06/12 17:12:02.845089 overlay_switch ->[2e:d4:1b:0c:6d:d5(ip-172-31-0-72)] using fastdp
INFO: 2018/06/12 17:12:02.845123 ->[172.31.0.72:6783|2e:d4:1b:0c:6d:d5(ip-172-31-0-72)]: connection added (new peer)
INFO: 2018/06/12 17:12:02.846566 ->[172.31.0.72:6783|2e:d4:1b:0c:6d:d5(ip-172-31-0-72)]: connection fully established
INFO: 2018/06/12 17:12:02.847314 ->[172.31.0.61:6783|1e:cd:af:bd:97:57(ip-172-31-0-61)]: connection ready; using protocol version 2
INFO: 2018/06/12 17:12:02.847377 overlay_switch ->[1e:cd:af:bd:97:57(ip-172-31-0-61)] using fastdp
INFO: 2018/06/12 17:12:02.847418 ->[172.31.0.61:6783|1e:cd:af:bd:97:57(ip-172-31-0-61)]: connection added (new peer)
INFO: 2018/06/12 17:12:02.848968 ->[172.31.0.61:6783|1e:cd:af:bd:97:57(ip-172-31-0-61)]: connection fully established
INFO: 2018/06/12 17:12:02.849603 EMSGSIZE on send, expecting PMTU update (IP packet was 60028 bytes, payload was 60020 bytes)
INFO: 2018/06/12 17:12:02.849969 EMSGSIZE on send, expecting PMTU update (IP packet was 60028 bytes, payload was 60020 bytes)
INFO: 2018/06/12 17:12:02.851132 sleeve ->[172.31.0.61:6783|1e:cd:af:bd:97:57(ip-172-31-0-61)]: Effective MTU verified at 8939
INFO: 2018/06/12 17:12:02.851256 sleeve ->[172.31.0.72:6783|2e:d4:1b:0c:6d:d5(ip-172-31-0-72)]: Effective MTU verified at 8939
INFO: 2018/06/12 17:12:02.899521 Discovered remote MAC 86:82:c4:45:81:36 at 2e:d4:1b:0c:6d:d5(ip-172-31-0-72)
INFO: 2018/06/12 17:12:02.907871 Discovered local MAC 06:23:ca:df:15:56
INFO: 2018/06/12 17:12:02.932900 Discovered remote MAC 3e:ba:ff:24:97:ee at 1e:cd:af:bd:97:57(ip-172-31-0-61)
INFO: 2018/06/12 17:12:03.028898 Discovered remote MAC 9e:82:94:11:7a:d1 at 1e:cd:af:bd:97:57(ip-172-31-0-61)
...
INFO: 2018/06/12 17:12:04.200661 GET /v1.17/containers/json?
INFO: 2018/06/12 17:12:04.204446 GET /v1.17/version
INFO: 2018/06/12 17:12:15.813463 POST /v1.17/images/create?fromImage=483324483244.dkr.ecr.us-east-1.amazonaws.com%2Fbenchmark-test%3Alatest
INFO: 2018/06/12 17:12:19.107058 POST /v1.17/images/create?fromImage=483324483244.dkr.ecr.us-east-1.amazonaws.com%2Ft11-test%3Alatest
INFO: 2018/06/12 17:12:29.064061 Discovered remote MAC 2a:7d:3c:3f:cb:18 at 2e:d4:1b:0c:6d:d5(ip-172-31-0-72)
INFO: 2018/06/12 17:12:29.864799 Discovered remote MAC ea:e1:2f:46:62:60 at 1e:cd:af:bd:97:57(ip-172-31-0-61)
INFO: 2018/06/12 17:12:29.880470 POST /v1.17/images/create?fromImage=redis%3Alatest
INFO: 2018/06/12 17:12:39.622569 GET /v1.17/images/redis/json
INFO: 2018/06/12 17:12:39.651429 POST /v1.18/containers/create?name=ecs-t3-RedisTask-B2EWFA5F6FNS-1-redis-c08da1d4a0e1fef44400
INFO: 2018/06/12 17:12:39.651547 Creating container with WEAVE_CIDR ""
INFO: 2018/06/12 17:12:39.870552 GET /v1.17/images/483324483244.dkr.ecr.us-east-1.amazonaws.com/t11-test:latest/json
INFO: 2018/06/12 17:12:40.037546 POST /v1.18/containers/create?name=ecs-t3-WorkerTask-XL6N3968UGI6-1-logos-delegate-poc-d0a1f4f0f1e497b0f601
INFO: 2018/06/12 17:12:40.037668 Creating container with WEAVE_CIDR ""
INFO: 2018/06/12 17:12:40.249422 GET /v1.17/containers/4d3395b7e7274567625ae8b1a194ee18dc113de001c5c176209b654800cbb335/json
INFO: 2018/06/12 17:12:40.249516 GET /v1.18/containers/4d3395b7e7274567625ae8b1a194ee18dc113de001c5c176209b654800cbb335/json
INFO: 2018/06/12 17:12:40.273801 Discovered remote MAC 4e:24:f2:12:d6:8f at 2e:d4:1b:0c:6d:d5(ip-172-31-0-72)
INFO: 2018/06/12 17:12:40.465593 POST /v1.18/containers/4d3395b7e7274567625ae8b1a194ee18dc113de001c5c176209b654800cbb335/start
INFO: 2018/06/12 17:12:40.468542 GET /v1.17/containers/767d482ef9a7650a5331aacb8695e528eb22324cd8514a66b47dd7a3614bd031/json
INFO: 2018/06/12 17:12:40.469072 GET /v1.18/containers/767d482ef9a7650a5331aacb8695e528eb22324cd8514a66b47dd7a3614bd031/json
INFO: 2018/06/12 17:12:40.483812 POST /v1.18/containers/767d482ef9a7650a5331aacb8695e528eb22324cd8514a66b47dd7a3614bd031/start
INFO: 2018/06/12 17:12:41.437980 Discovered remote MAC b6:ad:98:65:5e:37 at 1e:cd:af:bd:97:57(ip-172-31-0-61)
INFO: 2018/06/12 17:12:41.835419 GET /v1.17/containers/4d3395b7e7274567625ae8b1a194ee18dc113de001c5c176209b654800cbb335/json
INFO: 2018/06/12 17:12:41.837174 Attaching container 4d3395b7e7274567625ae8b1a194ee18dc113de001c5c176209b654800cbb335 with WEAVE_CIDR "" to weave network
INFO: 2018/06/12 17:12:41.837431 GET /v1.17/containers/4d3395b7e7274567625ae8b1a194ee18dc113de001c5c176209b654800cbb335/stats?stream=true
INFO: 2018/06/12 17:12:42.124725 GET /v1.17/containers/767d482ef9a7650a5331aacb8695e528eb22324cd8514a66b47dd7a3614bd031/json
INFO: 2018/06/12 17:12:42.126365 Attaching container 767d482ef9a7650a5331aacb8695e528eb22324cd8514a66b47dd7a3614bd031 with WEAVE_CIDR "" to weave network
INFO: 2018/06/12 17:12:42.127986 GET /v1.17/containers/767d482ef9a7650a5331aacb8695e528eb22324cd8514a66b47dd7a3614bd031/stats?stream=true
INFO: 2018/06/12 17:12:44.571840 Discovered local MAC be:d8:49:e9:cf:b0
INFO: 2018/06/12 17:12:44.585709 [nameserver 66:fb:9a:bf:54:f0] adding entry for 767d482ef9a7650a5331aacb8695e528eb22324cd8514a66b47dd7a3614bd031: logos-delegate-poc.weave.local. -> 10.46.0.1
INFO: 2018/06/12 17:12:44.586430 GET /v1.18/containers/767d482ef9a7650a5331aacb8695e528eb22324cd8514a66b47dd7a3614bd031/json
INFO: 2018/06/12 17:12:44.621194 Discovered local MAC 1a:2e:2a:6e:cc:ab
INFO: 2018/06/12 17:12:44.641861 [nameserver 66:fb:9a:bf:54:f0] adding entry for 4d3395b7e7274567625ae8b1a194ee18dc113de001c5c176209b654800cbb335: redis.weave.local. -> 10.46.0.0
INFO: 2018/06/12 17:12:44.643017 GET /v1.18/containers/4d3395b7e7274567625ae8b1a194ee18dc113de001c5c176209b654800cbb335/json
INFO: 2018/06/12 17:12:51.889357 Discovered remote MAC e2:e0:37:21:38:bc at 1e:cd:af:bd:97:57(ip-172-31-0-61)
INFO: 2018/06/12 17:12:52.174289 GET /v1.17/images/483324483244.dkr.ecr.us-east-1.amazonaws.com/benchmark-test:latest/json
INFO: 2018/06/12 17:12:52.177041 POST /v1.18/containers/create?name=ecs-t3-MasterTask-X91HLAQW0OZK-1-master-8ac1c1feacf59cc87e00
INFO: 2018/06/12 17:12:52.177175 Creating container with WEAVE_CIDR ""
INFO: 2018/06/12 17:12:52.489051 GET /v1.18/containers/d8e7f29d7ca1801e665b3aeadf412c6aa9f16d5c1475532890ce5d8e0a12c770/json
INFO: 2018/06/12 17:12:52.489754 GET /v1.17/containers/d8e7f29d7ca1801e665b3aeadf412c6aa9f16d5c1475532890ce5d8e0a12c770/json
INFO: 2018/06/12 17:12:52.491181 POST /v1.18/containers/d8e7f29d7ca1801e665b3aeadf412c6aa9f16d5c1475532890ce5d8e0a12c770/start
INFO: 2018/06/12 17:12:52.758689 Attaching container d8e7f29d7ca1801e665b3aeadf412c6aa9f16d5c1475532890ce5d8e0a12c770 with WEAVE_CIDR "" to weave network
INFO: 2018/06/12 17:12:52.759086 GET /v1.17/containers/d8e7f29d7ca1801e665b3aeadf412c6aa9f16d5c1475532890ce5d8e0a12c770/json
INFO: 2018/06/12 17:12:52.760762 GET /v1.17/containers/d8e7f29d7ca1801e665b3aeadf412c6aa9f16d5c1475532890ce5d8e0a12c770/stats?stream=true
INFO: 2018/06/12 17:12:52.881949 Discovered local MAC 66:1d:53:6f:ce:1d
INFO: 2018/06/12 17:12:52.892535 [nameserver 66:fb:9a:bf:54:f0] adding entry for d8e7f29d7ca1801e665b3aeadf412c6aa9f16d5c1475532890ce5d8e0a12c770: master.weave.local. -> 10.46.0.2
INFO: 2018/06/12 17:12:52.893320 GET /v1.18/containers/d8e7f29d7ca1801e665b3aeadf412c6aa9f16d5c1475532890ce5d8e0a12c770/json
INFO: 2018/06/12 17:12:54.081592 Discovered remote MAC 62:99:97:1c:4b:79 at 2e:d4:1b:0c:6d:d5(ip-172-31-0-72)
```

```
]# uname -a
Linux ip-172-31-0-170 4.9.43-17.38.amzn1.x86_64 #1 SMP Thu Aug 17 00:20:39 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
[root@ip-172-31-0-170 /]# docker version
Client:
 Version:      17.03.2-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   7392c3b/17.03.2-ce
 Built:        Wed Aug  9 22:45:09 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.2-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   7392c3b/17.03.2-ce
 Built:        Wed Aug  9 22:45:09 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by bboreham** [comment]
This issue was already reported at #3136 and fixed at #3138.
The issue with the AMI is that it installs Weave Net 2.0.4, and the fix went into 2.0.5.

**Comment by bboreham** [comment]
I've updated the AMIs - see https://github.com/weaveworks/scope/pull/3226
